### PR TITLE
7.0 bug compute qty same uom

### PR DESCRIPTION
**Pull request by legalsylvain**
Case:
- Create a new unit 6kg based on kg product uom category with factor 6 and rounding 1.0;
- Call the function _compute_qty of product_uom Model with the folllowing parameters:
  - from_unit = to_unit = 6kg
  - qty = 1
- Expected value: 1 (the input value, because there is no conversion to do).
- Return value: 0. (depending of rounding setting).

The case is written in a TransactionCase in test_uom.py

**Classical User Case**:
when an Delivery order is validated, the function _compute_qty is called, even if the units are identical.

### CODE REVIEW

**Comment by legalsylvain** [comment]
You can review directly odoo Branch if you want.
https://github.com/odoo/odoo/pull/3693

**Comment by pedrobaeza** [comment]
Good escape condition!

:+1:

**Comment by legalsylvain** [comment]
FYI, this PR has a second :+1: in the Odoo PR. https://github.com/odoo/odoo/pull/3693
Regards.

**Comment by lmignon** [comment]
:+1: And thank you for the unittest!

**Comment by gfcapalbo** [comment]
A freshly updated openerp7 (today), passes the test with no need for the fix. 
The fix is an extra check , and I can't think of any cases where it could cause unforseen problems,
so i'll :+1: 
It would be interesting to know if your current setup (without merging this PR) still has the bug.

**Comment by hbrunn** [comment]
:-1: I disagree: It should be safe to assume that _compute_qty respects the UOM's rouding. Probably there is some code that uses it just for that.
If you need three nails and they only come by the dozen, you'll have to order the dozen. In my opinion, _compute_qty(dozen, .25, dozen) should return 1 (assuming we set its rounding to 1.0).
 Did you check higher numbers? To me, what you describe sounds more like an issue with float_round.

File: addons/product/tests/test_uom.py [hunk 1]
```diff
@@ -20,6 +20,27 @@ def test_10_conversion(self):
...
 
+    def test_11_conversion(self):
+        """compute qty must return qty if uom_from_id and uom_to_id
+        are identical."""
+        cr, uid = self.cr, self.uid
+        categ_kg_id = self.imd.get_object_reference(
+            cr, uid, 'product', 'product_uom_categ_kgm')[1]
+        uom_6kg_id = self.uom.create(cr, uid, {
+            'name': '6 kg',
+            'factor': 6,
```
Comment (hbrunn) [inline addons/product/tests/test_uom.py:31]: that should be factor_inv, right?

**Comment by hbrunn** [comment]
The original issue @legalsylvain describes is probably fixed in https://github.com/OCA/OCB/commit/fc85a7ee5c33eadc22652b6b26478c320e35b283, so I propose to close this and https://github.com/odoo/odoo/issues/3693

**Comment by hbrunn** [comment]
as explained in https://github.com/OCA/OCB/pull/91#issuecomment-74956627, I close this

**Pull request closed without merging**